docs/source/model/dependencies.rst

```diff
@@ -0,0 +1,453 @@
+
+::
+
+    my_project/
+    |── train.py
+    └── src/
+        └──  utils.py
+
+Then the following model code does **not** work:
+
+.. code-block:: python
+
+    class MyModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input):
+            from src.utils import my_func
+
+            # .. your prediction logic
+            return prediction
+
+
+    with mlflow.start_run() as run:
+        mlflow.pyfunc.log_model(
+            python_model=MyModel(),
+            artifact_path="model",
+            code_paths=[
+                "src/utils.py"
+            ],  # the file will be saved at code/utils.py not code/src/utils.py
+        )
+
+    # => Model serving will fail with ModuleNotFoundError: No module named 'src'
+
+This limitation is due to how MLflow saves and loads the specified files/directories. When it copies the specified files or directries in ``code/`` target,
+it does **not** preserve the relative path for them. For instance, in the above example, MLflow will copy ``utils.py`` to ``code/utils.py``, not
+``code/src/utils.py``. As a result, it has to be imported as ``from utils import my_func``, instead of ``from src.utils import my_func``.
+However, this may not be pleasant as the import path is different from the original training script.
+
+One workaround for this issue is to use the parent directory instead, which means doing ``code_path=["src"]`` in this example.
+This way, MLflow will copy the entire ``src/`` directory under ``code/`` and your model code will be able to import ``src.utils``.
+
+.. code-block:: python
+
+    class MyModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input):
+            from src.utils import my_func
+
+            # .. your prediction logic
+            return prediction
+
+
+    with mlflow.start_run() as run:
+        mlflow.pyfunc.log_model(
+            python_model=model,
+            artifact_path="model",
+            code_paths=["src"],  # the whole /src directory will be saved at code/src
+        )
+
+    # => This will work
+
+.. warning::
+
+    By the same reason, ``code_path`` option doesn't handle the relative import like ``code_path=["../src"]``.
+
+
+.. _validating-environment-for-prediction:
+
+Validating Environment for Prediction
+-------------------------------------
+
+Validating your model before deployment is a critical step to ensure production readiness.
+MLflow provides a few ways to test your model locally, either in a virtual environment or a Docker container.
+If you find any dependency issues during validation, please follow the guidance in :ref:`How to fix dependency errors when serving my model? <how-to-fix-dependency-errors-in-model>`
+
+Testing offline prediction with a virtual environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You can use MLflow **predict** API via CLI or Python to make test predictions with your model.
```

Comment:
Can we specify that this is the MLflow **Models** predict API and also mention which version of MLflow this is available in (as it's so new, users who haven't upgraded yet might want to use it, only to find that it doesn't exist in their version of the client)